### DISCUSSION TITLE
Can't use kubeconfig from GKE connectionSecret

### DISCUSSION THREAD

**Discussion by calavelas** [crossplane/crossplane#4488] [Q&A] [answered]
Hi!

I've been struggling to do a composite cluster on GCP,
I can create GKE cluster and be able to save connection secret (kubeconfig) to create helm providerConfig

My problems is I cant connect to the GKE cluster with that providerConfig and I think it due to this change in GKE
https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke

Is there any workaround to connect to cluster and install ArgoCD for cluster bootstraping?

**Comment by deggja** [comment]
If I understand you correctly, I think you are trying to achieve the same as me. Might be related to the following PR: https://github.com/crossplane-contrib/provider-argocd/pull/66
That is, if you are trying to add the GKE cluster to Argo automatically with provider-argocd.

**Reply by calavelas** [reply]
I've seen that PR too, What I tried to do is using helm to install Argo (which doesnt works due to GKE auth).

have you tried @tpickett method on that post https://github.com/crossplane-contrib/provider-argocd/pull/66
(ArgoCD with Workload Identity) I think he pulled that off but I haven't go time to try it myself. Last I checked ArgoCD 
already supports GKE Auth officially https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/#clusters

**Comment by turkenh** [comment, marked as answer]
@calavelas did you already check https://github.com/upbound/platform-ref-gcp ? 

It also creates a GKE cluster and installs some helm releases with the help of a ProviderConfig as defined here https://github.com/upbound/platform-ref-gcp/blob/main/package/cluster/gke/composition.yaml#L177

**Reply by calavelas** [reply]
I haven't seen this one before, Will try this out

Thank!
